src/pages/SupplyRequisitionPage.js

```diff
@@ -0,0 +1,334 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import autobind from 'react-autobind';
+import { View } from 'react-native';
+
+import { GenericPage } from './GenericPage';
+import globalStyles from '../globalStyles';
+import { formatDate, parsePositiveInteger, sortDataBy } from '../utilities';
+import { buttonStrings, modalStrings, pageInfoStrings, tableStrings } from '../localization';
+import {
+  PageButton,
+  PageInfo,
+  PageContentModal,
+  TextEditor,
+} from '../widgets';
+
+const DATA_TYPES_SYNCHRONISED = ['RequisitionItem', 'Item', 'ItemBatch'];
+const MODAL_KEYS = {
+  COMMENT_EDIT: 'commentEdit',
+  ITEM_SELECT: 'itemSelect',
+  MONTHS_SELECT: 'monthsSelect',
+};
+
+export class SupplyRequisitionPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalKey: null,
+      modalIsOpen: false,
+      selection: [],
+    };
+    this.dataFilters = {
+      searchTerm: '',
+      sortBy: 'itemName',
+      isAscending: true,
+    };
+    autobind(this);
+  }
+
+  /**
+   * Respond to the user editing the number in the required quantity column
+   * @param  {string} key             Should always be 'requiredQuantity'
+   * @param  {object} requisitionItem The requisition item from the row being edited
+   * @param  {string} newValue        The value the user entered in the cell
+   * @return {none}
+   */
+  onEndEditing(key, requisitionItem, newValue) {
+    // This will update associated CustomerInvoice if one exists or if linked
+    // CustomerInvoice does not exist, suppliedQuantity will not be updated
+    if (key !== 'suppliedQuantity') return;
+    const { database } = this.props;
+    database.write(() => {
+      const transactionItem = requisitionItem.linkedTransactionItem;
+      if (!transactionItem) return;
+      transactionItem.setTotalQuantity(database, parsePositiveInteger(newValue),
+                                                 transactionItem.availableQuantity);
+      requisitionItem.suppliedQuantity = transactionItem.totalQuantity;
+      database.save('TransactionItem', transactionItem);
+      database.save('RequisitionItem', requisitionItem);
```

Comment:
As per comment above where setTotalQuantity is used in CustomerInvoicePage -> before I refactored.

My original comment wasn't very good ( I misread your comment requisition item -> transaction item ). And yes, will leave 'sync' out of save and update talk and use them all the time dot notation assignment is used.